docs/faq/ImmutableData.md

```diff
@@ -1,245 +1,145 @@
-const newState = state.setIn(['prop1’], fromJS(newObj)); // <-- newObj is now an
-    // Immutable.JS Map
-```
-
-#### Use Immutable.JS everywhere except your dumb components
-
-Using Immutable.JS everywhere keeps your code performant. Use it in your smart components, your selectors, your sagas or thunks, action creators, and especially your reducers. 
-
-Do NOT, however, use Immutable.JS in your dumb components.
+let record = { id: 0 };
 
-#### Limit your use of `toJS()`
+const getNewRecord = () => {
+  ++record.id;
+  return record;
+}
 
-`toJS()` is an expensive function and negates the purpose of using Immutable.JS. Avoid its use.
+const mapStateToProps = (state) => ({
+	// The object returned from getNewRecord() is always
+	// the same object, so this connected
+	// component will never re-render
+	record: getNewRecord()
+});
 
-#### Your selectors should return Immutable.JS objects
 
-Always.
+const a = mapStateToProps();
+const b = mapStateToProps();
 
-#### Use Immutable.JS objects in your Smart Components
+a.record === b.record
+//> true
 
-Smart components that access the store via React Redux’s `connect` function must use the Immutable.JS values returned by your selectors. 
-
-#### Never use `toJS()` in `mapStateToProps`
-
-Converting an Immutable.JS object to a JavaScript object using `toJS()` will return a new object every time. If you use do this in `mapSateToProps`, you will cause the component to believe that the object has changed every time the state tree changes, and so trigger an unnecessary re-render.
+```
 
-#### Never use Immutable.JS in your Dumb Components
 
-Your dumb components should rely solely on JavaScript. Using Immutable.JS in your components adds an extra dependency and stops them from being portable.
 
-#### Use a Higher Order Component to convert your Smart Component’s Immutable.JS props to your Dumb Component’s JavaScript props
+### How does immutability enable a shallow check to detect object mutations?
+If an object is immutable, any changes that need to be made to it within a function must be made to a _copy_ of the object. 
 
-Something needs to map the Immutable.JS props in your Smart Component to the pure JavaScript props used in your Dumb Component. That something is a Higher Order Component (HOC) that simply takes the Immutable.JS props from your Smart Component, and converts them using `toJS()` to plain JavaScript props, which are then passed to  your Dumb Compnent.
+This mutated copy is a _separate_ object from that passed into the function, and so when it is returned, a shallow check will identify it as being a different object from that passed in, and so return false.
 
-Here is an example of such a HOC:
+### What issues can immutability cause with Redux?
+If you make a copy of the `state` object that's passed into a reducer, and return this copy, then Redux will think that the state has been updated, as you're returning an entirely different object from the `state` object that was passed in.
```

Comment:
Maybe clarify this section a bit? I think the key here is that copies should only be made of an object/array _if_ something has actually changed - doing something like `let newState = {...state}` at the start of a reducer could cause excess re-rendering even if you hit the "default" case in a switch statement.